### PR TITLE
Always use static ports for client load-balancers

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

This fixes an issue where RKE2 kube-proxy daemonset pods were failing to communicate with the apiserver after RKE2 was restarted because the load-balancer used a different port every time it started up.

This also changes the apiserver load-balancer port to be 1 below the supervisor port instead of 1 above it. This makes the apiserver port consistent at 6443 across servers and agents on RKE2.

I also had to pull in a few other fixes to etcd-only mode as well - the new CLI flag wasn't actually being used which caused problems for the load-balancer port fix, and the server would crash or fail to connect to other servers under certain circumstances.

#### Types of Changes ####

* load-balancer ports
* misc etcd-only stuff

#### Verification ####

See steps in linked issue

#### Linked Issues ####

rancher/rke2#573

#### Further Comments ####

Comparison of port usage before/after this change:

**Before**
```
2379 k3s/rke2 etcd client port on server if running etcd
     k3s/rke server LB tunnel to etcd client port on server if not running etcd
2380 k3s/rke2 server etcd peer port if running etcd
6443 k3s supervisor + external apiserver on server if running apiserver
     rke2 apiserver on server if running apiserver
6444 k3s local apiserver if running apiserver
     k3s LB tunnel to supervisor + external apiserver on server if not running apiserver
     rke2 LB tunnel to supervisor on server if not running apiserver
6445 rke2 tunnel to apiserver on server if not running apiserver
9345 rke2 supervisor on server
XXXX k3s LB tunnel to supervisor + external apiserver on agent (randomly selected on startup)
     rke2 LB tunnel to supervisor on agent (randomly selected on startup)
YYYY rke2 LB tunnel to apiserver on agent (randomly selected on startup)
```
**After**
```
2379 k3s/rke2 etcd client port on server if running etcd
     k3s/rke2 LB tunnel to etcd client port on server if not running etcd
2380 k3s/rke2 etcd peer port on server if running etcd
6443 k3s supervisor + external apiserver on server if running apiserver
     rke2 apiserver on server if running apiserver
     rke2 LB tunnel to apiserver on agent, and server if not running apiserver
6444 k3s local apiserver on server if running apiserver
     k3s LB tunnel to supervisor + external apiserver on agent, and server if not running apiserver
     rke2 LB tunnel to supervisor on agent, and server if not running apiserver
9345 rke2 supervisor on server
```

### CODE REVIEW

**Review by galal-hussein — APPROVED** [review]
LGTM, thanks for the comments on the proxy code

File: pkg/cli/cmds/agent.go [hunk 1]
```diff
@@ -167,11 +167,11 @@ var (
 	}
 	LBServerPort = cli.IntFlag{
 		Name:        "lb-server-port",
-		Usage:       "(agent/node) Internal Loadbalancer port",
+		Usage:       "(agent/node) Local port for supervisor tunnel load-balancer. If the supervisor and apiserver are not colocated; an additional port 1 below this one will also be used for the apiserver tunnel.",
```
Comment (Oats87) [inline pkg/cli/cmds/agent.go:170]: I can't help but feel that the description here is a little bit vague, I can understand it as `1 below this one` i.e. 1 below 6444 is 6443 but I cannot say the same for everyone.

**Review by Oats87 — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

File: pkg/cli/cmds/agent.go (hunk 1 above)
Comment (brandond) [reply pkg/cli/cmds/agent.go]: Just tweaked it a bit, is that any better?

Comment (Oats87) [reply pkg/cli/cmds/agent.go]: Yeah, that looks great! Thanks

**Review by galal-hussein — APPROVED** [review]
LGTM, thanks

**Review by briandowns — APPROVED** [review]

**Pull request merged**